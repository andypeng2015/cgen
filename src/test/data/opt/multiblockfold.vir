module foo

export function w $foo(w %x) {
@start:
  %x = add.w %x, 1_w
  jmp @a
@a:
  %x = add.w %x, 1_w
  jmp @b
@b:
  %x = add.w %x, 1_w
  jmp @c
@c:
  %x = sub.w %x, 1_w
  ret %x
}
