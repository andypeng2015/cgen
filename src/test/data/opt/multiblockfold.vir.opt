module foo

export function w $foo(w %x) {
@2:
  %1 = add.w %x, 0x2_w ; @11
  ret %1
}
