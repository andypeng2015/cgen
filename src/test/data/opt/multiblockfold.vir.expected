module foo

export function w $foo(w %x) {
@0:
  %0 = add.w %x, 0x2_w ; @8
  ret %0
}
